README.md

```diff
@@ -1,3 +1,224 @@
+
+**Chunking** addresses the issue of deploying contracts that exceed Sway's 100KB contract size limit. It splits the oversized contract into multiple blobs and uses a loader contract. When the loader contract is called upon, it automatically loads the blobs into memory, allowing the contract to function as normal.
+
+Both features are abstracted from developers and work out of the box with the latest version of `forc`. Below is a reference on how to use `Proxies`, `Chunking`, and how to combine them.
+
+## Prerequisites
+
+If you already have `fuelup` installed, run the commands below to make sure you are on the most up-to-date toolchain.
+
+```sh
+fuelup self update
+fuelup update
+fuelup default latest
+```
+
+At the time of writing you should be on `v0.63.3` or above
+
+```sh
+fuelup show
+```
+
+The command above should yield a similar output below
+
+```sh
+active toolchain
+----------------
+latest-aarch64-apple-darwin (default)
+  forc : 0.63.3
+    - forc-client
+      - forc-deploy : 0.63.3
+      - forc-run : 0.63.3
+    - forc-crypto : 0.63.3
+    - forc-debug : 0.63.3
+    - forc-doc : 0.63.3
+    - forc-fmt : 0.63.3
+    - forc-lsp : 0.63.3
+    - forc-tx : 0.63.3
+    - forc-wallet : 0.9.0
+  fuel-core : 0.34.0
+  fuel-core-keygen : 0.34.0
+
+fuels versions
+--------------
+forc : 0.66.1
+forc-wallet : 0.66.0
+```
+
+## Proxy
+
+> DO NOT ATTEMPT TO USE THE SRC-14 STANDARD OR UPGRADABILITY DIRECTLY YOURSELF TO AVOID COLLISION CATASTROPHES. THEY ARE FOR EDUCATIONAL PURPOSES ONLY. 
+
+> Please refer to [using the Proxy](#using-the-proxy) section to safely incoporate proxies into your project.
+
+
+#### [SRC-14 Simple Upgradeable Proxies](https://docs.fuel.network/docs/sway-standards/src-14-simple-upgradeable-proxies/)
+
+The SRC-14 standard is similar to upgradeable proxies in Ethereum like UUPS, where an implementation/target contract is stored in storage of the proxy contract and all calls are delegated to it. 
+
+In FuelVM, this behavior is implemented using the `LDC` instruction, which Sway's `std::execution::run_external` leverages to mimic EVM's `delegatecall`, allowing execution of instructions from another contract while retaining the original contract's storage context.
+
+Essential functions to note are `set_proxy_target(new_target: ContractId)` and `proxy_target()` as mentioned above.
+
+```sway
+abi SRC14 {
+    #[storage(read, write)]
+    fn set_proxy_target(new_target: ContractId);
+    #[storage(read)]
+    fn proxy_target() -> Option<ContractId>;
+}
+ 
+abi SRC14Extension {
+    #[storage(read)]
+    fn proxy_owner() -> State;
+}
+```
+
+#### [Upgradable Libraries](https://docs.fuel.network/docs/sway-libs/upgradability/)
+
+To use the SRC-14 standard along with the upgradability library, you simply need to import the necessary modules as shown below:
```

Comment:
This will need an additional section as the upgaradability lib makes an assumption about the proxy_owner key:

could you add a bit that mentions that the lib assumes the `proxy_owner` will be stored at `sha256("storage_SRC14_1")` like so :
```sway
/// The storage slot to store the proxy owner State.
///
/// Value is `sha256("storage_SRC14_1")`.
pub const PROXY_OWNER_STORAGE: b256 = 0xbb79927b15d9259ea316f2ecb2297d6cc8851888a98278c0a2e03e1a091ea754;
```

as such a proxy contract that uses this lib and follows src14 should setup the storage values of `target` and `proxy_owner` like so:
```sway
storage {
    SRC14 {
        /// The [ContractId] of the target contract.
        ///
        /// # Additional Information
        ///
        /// `target` is stored at sha256("storage_SRC14_0")
        target in 0x7bb458adc1d118713319a5baa00a2d049dd64d2916477d2688d76970c898cd55: Option<ContractId> = None,
        /// The [State] of the proxy owner.
        ///
        /// # Additional Information
        ///
        /// `proxy_owner` is stored at sha256("storage_SRC14_1")
        proxy_owner in 0xbb79927b15d9259ea316f2ecb2297d6cc8851888a98278c0a2e03e1a091ea754: State = State::Uninitialized,
    },
}
```